src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -207,6 +208,16 @@ case class MergeIntoCommand(
 
   @transient private lazy val sc: SparkContext = SparkContext.getOrCreate()
   @transient private lazy val targetDeltaLog: DeltaLog = targetFileIndex.deltaLog
+  @transient private lazy val targetOutputAttributesMap: Map[String, Attribute] = {
```

Comment:
Add docs explaining what this map contains and is for.

Reply:
so the only place this map is used is in `getTargetOutputCols`, which uses to map to generate a different collection. why not just cache that final collection.. and not the intermediate map?

Reply:
I do not know so much in details the code base so I just tried to reduce changes without guidance.
If I understood correctly you would like to cache the result of `getTargetOutputCols` which seems reasonable to me, but I have a question: can we assume schema in the metadata of the transaction will never change and only one transaction will be used?

Reply:
Docs added and code rebased.